src/content/en/updates/2017/11/overscroll-behavior.md

```diff
@@ -0,0 +1,370 @@
+history.
+
+For this app, it's more appropriate to have scrolls that originate within the
+chatbox stay within the chat. We can make that happen by adding
+`overscroll-behavior: contain` to the element that holds the chat messages:
+
+```css
+#chat .msgs {
+  overflow: auto;
+  overscroll-behavior: contain;
+  height: 300px;
+}
+```
+
+Essentially, we're creating a logical separation between the chatbox's scrolling
+context and the main page. The end result is that the main page stays put when
+the user reaches the top/bottom of the chat history. Scrolls that start in the
+chatbox do not propagate out.
+
+### The page overlay scenario {: #overlay }
+
+Another variation of the "underscroll" scenario is when you see content
+scrolling behind a **fixed position overlay**. A dead giveaway
+`overscroll-behavior` is in order! The browser is trying to be helpful but
+it ends up making the site look buggy.
+
+**Example** - modal with and without `overscroll-behavior: contain`:
+
+<figure class="flex">
+  <div class="flex1">
+    <a href="/web/updates/images/2017/11/overscroll-behavior/modal-off.mp4"
+       target="_blank">
+      <video src="/web/updates/images/2017/11/overscroll-behavior/modal-off.mp4"
+             autoplay loop height="290"></video>
+    </a>
+    <figcaption>
+      <b>Before</b>: page content scrolls beneath overlay.
+    </figcaption>
+  </div>
+  <div class="flex1">
+    <a href="/web/updates/images/2017/11/overscroll-behavior/modal-on.mp4"
+       target="_blank">
+      <video src="/web/updates/images/2017/11/overscroll-behavior/modal-on.mp4"
+             autoplay loop height="290"></video>
+    </a>
+    <figcaption>
+      <b>After</b>: page content doesn't scroll beneath overlay.
+    </figcaption>
+  </div>
+</figure>
+
+## Disabling pull-to-refresh  {: #disablp2r }
+
+**Turning off the pull-to-refresh action is a single line of CSS**. Just prevent
+scroll chaining on the entire viewport-defining element. In most cases, that's
+`<html>` or `<body>`:
+
+```css
+body {
+  /* Disables pull-to-refresh but allows overscroll glow effects. */
+  overscroll-behavior-y: contain;
+}
+```
+
+With this simple addition, we fix the double pull-to-refresh animations in
+the [chatbox demo](https://ebidel.github.io/demos/chatbox.html) and can
+instead, implement a custom effect which uses a neater loading animation. The
+entire inbox also blurs as the inbox refreshes:
+
+<figure class="flex">
+  <div class="flex1">
+    <video src="/web/updates/images/2017/11/overscroll-behavior/chatbox-double-refresh.mp4"
+           autoplay loop height="225"></video>
+    <figcaption>Before</figcaption>
+  </div>
+  <div class="flex1">
+    <video src="/web/updates/images/2017/11/overscroll-behavior/chatbox-double-refresh-fix.mp4"
+           autoplay loop height="225"></video>
+    <figcaption>After</figcaption>
+  </div>
+</figure>
+
+Here's a snippet of the
+[full code](https://github.com/ebidel/demos/blob/master/chatbox.html):
```

Comment:
Shouldn't this be on a chrome/google git repo?

Reply:
Perhaps. But I tend to put one off mini examples on a gist or that /demos repo. Been around for a while.

Reply:
Ok - we'll leave it for now, but will add that as something to review.